src/components/menu-item/menu-item.tsx

```diff
@@ -0,0 +1,469 @@
+  @Prop({ reflect: true }) iconStart: string;
+
+  /** Specifies an icon to display at the end of the component. */
+  @Prop({ reflect: true }) iconEnd: string;
+
+  /** Displays the `iconStart` and/or `iconEnd` as flipped when the element direction is right-to-left (`"rtl"`). */
+  @Prop() iconFlipRtl: FlipContext;
+
+  /** Specifices accessible name for the component.*/
+  @Prop() label!: string;
+
+  /**
+   * Defines the relationship between the `href` value and the current document.
+   *
+   * @mdn [rel](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/rel)
+   */
+  @Prop({ reflect: true }) rel: string;
+
+  /**
+   * Specifies where to open the linked document defined in the `href` property.
+   *
+   * @mdn [target](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-target)
+   */
+  @Prop({ reflect: true }) target: string;
+
+  /** Specifies the text to display.*/
+  @Prop({ reflect: true }) text: string;
+
+  /** When `true`, the menu item will display any slotted `calcite-menu-item` in an open overflow menu.*/
+  @Prop({ mutable: true, reflect: true }) open = false;
+
+  /**
+   * @internal
+   */
+  @Prop({ reflect: true }) layout: Layout;
+
+  /**
+   * @internal
+   */
+  @Prop() isTopLevelItem = false;
+
+  /**
+   * @internal
+   */
+  @Prop() topLevelMenuLayout: Layout;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private State/Props
+  //
+  //--------------------------------------------------------------------------
+
+  private anchorEl: HTMLAnchorElement;
+
+  private dropDownActionEl: HTMLCalciteActionElement;
+
+  private isFocused: boolean;
+
+  @State() hasSubMenu = false;
+
+  @State() subMenuItems: HTMLCalciteMenuItemElement[];
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  //--------------------------------------------------------------------------
+
+  /** Sets focus on the component. */
+  @Method()
+  async setFocus(): Promise<void> {
+    await componentLoaded(this);
+    this.anchorEl.focus();
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+  /** @internal */
+  @Event({ cancelable: true }) calciteInternalMenuItemKeyEvent: EventEmitter<MenuItemCustomEvent>;
+
+  /** Emits when user selects the component. */
+  @Event({ cancelable: false }) calciteMenuItemSelect: EventEmitter<void>;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Event Listeners
+  //
+  //--------------------------------------------------------------------------
+  @Listen("click", { target: "window" })
+  handleClickOut(event: Event): void {
+    if (
+      this.topLevelMenuLayout !== "vertical" &&
+      this.hasSubMenu &&
+      this.open &&
+      !this.el.contains(event.target as Element)
```

Comment:
You may need to use composedPath here since this won't work if the target is inside a shadowDom.